doc: matter: add figure for factory data partition (#10744)
nrfconnect/sdk-nrf: doc/nrf/protocols/matter/end_product/bootloader.rst

By default, the ``factory_data`` partition is write-protected with the :ref:`fprotect_readme` driver (``fprotect``).
The hardware limitations require that the write-protected areas are aligned to :kconfig:option:`CONFIG_FPROTECT_BLOCK_SIZE`.
For this reason, to effectively implement ``fprotect``, make sure that the partition layout of the application meets the following requirements:

* The ``factory_data`` partition is placed right after the ``app`` partition in the address space.

> * The ``settings_storage`` partition size must be a multiple of :kconfig:option:`CONFIG_FPROTECT_BLOCK_SIZE`, which may differ depending on the SoC in use.

Reviewer: ```suggestion
* The ``settings_storage`` partition size is a multiple of :kconfig:option:`CONFIG_FPROTECT_BLOCK_SIZE`, which may differ depending on the SoC in use.
```
to use a consistent modality.

> See the following figure and make sure to check the :ref:`ug_matter_hw_requirements_layouts` to make sure your implementation is correct.

Reviewer: ```suggestion
See the following figure and check the :ref:`ug_matter_hw_requirements_layouts` to make sure your implementation is correct.
```
to avoid double "make sure" :)
